### PR TITLE
add progress bar to FastICA

### PR DESCRIPTION
**Pull request by sunalbert**
add progress bar to FastICA algorithm.

### CODE REVIEW

File: src/shogun/converter/ica/FastICA.cpp [hunk 1]
```diff
@@ -128,9 +128,9 @@ CFeatures* CFastICA::apply(CFeatures* features)
 
 	W = sym_decorrelation(W);
 
-	int iter = 0;
 	float64_t lim = tol+1;
-	while (lim > tol && iter < max_iter)
+	for (auto i : progress(
+	         range(0, max_iter), "PROGRESS: ", UTF8, [&] { return lim > tol; }))
```
Comment (iglesias) [inline src/shogun/converter/ica/FastICA.cpp]: Minor, for consistency, indent and break lines like in the doc reference you've cited.

Comment (sunalbert) [reply src/shogun/converter/ica/FastICA.cpp]: Thanks @iglesias . The style has been modified.

**Comment by iglesias** [comment]
Sorry that I did not mention it earlier, I have noticed only now the number of commits. Can you squash the 4 into 1? Thanks!

File: src/shogun/converter/ica/FastICA.cpp [hunk 2]
```diff
@@ -130,7 +130,11 @@ CFeatures* CFastICA::apply(CFeatures* features)
 
 	float64_t lim = tol+1;
 	for (auto i : progress(
-	         range(0, max_iter), "PROGRESS: ", UTF8, [&] { return lim > tol; }))
+	    range(0, max_iter),
+		"PROGRESS: ",
```
Comment (iglesias) [inline src/shogun/converter/ica/FastICA.cpp]: Put this line and the two below at the same indentation than range, like in the doc we were referring to.

**Comment by sunalbert** [comment]
Thanks @iglesias I will change the code style and squash all the commits into one.

**Comment by karlnapf** [comment]
@iglesias we can now squash with github, so people can just keep on adding commits to PRs (better to see trajectory, history, and preserving discussions (at least for small changes that make sense to group into a single commit)

File: src/shogun/converter/ica/FastICA.cpp [hunk 3]
```diff
@@ -128,9 +128,13 @@ CFeatures* CFastICA::apply(CFeatures* features)
 
 	W = sym_decorrelation(W);
 
-	int iter = 0;
 	float64_t lim = tol+1;
-	while (lim > tol && iter < max_iter)
+	for (auto i : progress(
+	    range(0, max_iter),
+	    "PROGRESS: ",
```
Comment (karlnapf) [inline src/shogun/converter/ica/FastICA.cpp]: @geektoni this seems a bit verbose. Isnt there a shorter version?

Comment (karlnapf) [reply src/shogun/converter/ica/FastICA.cpp]: Sorry! I just only realise this is the advanced way of using it

Comment (karlnapf) [reply src/shogun/converter/ica/FastICA.cpp]: @geektoni Could we have a way to pass lambda functions that does not require to pass the string and the encoding?
I.e. something as compact as the `for (auto i: progress(range(0, 10)))` but with a custom stopping condition?

Comment (geektoni) [reply src/shogun/converter/ica/FastICA.cpp]: #4240

Comment (karlnapf) [reply src/shogun/converter/ica/FastICA.cpp]: @sunalbert could you rebase and use the new function?

Comment (sunalbert) [reply src/shogun/converter/ica/FastICA.cpp]: Ok, Thanks @iglesias @karlnapf

Comment (iglesias) [reply src/shogun/converter/ica/FastICA.cpp]: Of course, you are welcome!

**Comment by karlnapf** [comment]
All good the error is not caused by you

**Pull request merged**